osquery/tables/system/tests/posix/apt_sources_tests.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <fstream>
+
+#include <osquery/tables/system/posix/apt_sources.h>
+
+#include <boost/filesystem.hpp>
+
+#include <gtest/gtest.h>
+
+namespace osquery {
+namespace tables {
+
+class AptSourcesImplTests : public testing::Test {};
+
+struct AptTestCase {
+  std::string input_line;
+  std::string base_uri;
+  std::string name;
+  std::string cache_filename;
+};
+
+TEST_F(AptSourcesImplTests, parse_apt_source_line) {
+  std::vector<AptTestCase> test_cases = {
+      // "deb " is stripped off.
+      {"[arch=amd64] https://pkg.osquery.io/deb deb main",
+       "https://pkg.osquery.io/deb",
+       "pkg.osquery.io/deb deb main",
+       "pkg.osquery.io_deb_dists_deb"},
+      // "dists" is not used as a section.
+      {"https://pkg.osquery.io apt/source/",
+       "https://pkg.osquery.io",
+       "pkg.osquery.io apt source",
+       "pkg.osquery.io_apt_source"},
+      // trailing slashes are stripped from URI.s
+      {"https://pkg.osquery.io/// apt/source/",
+       "https://pkg.osquery.io///",
+       "pkg.osquery.io apt source",
+       "pkg.osquery.io_apt_source"},
```

Comment:
I would add a case for `{"https://", "https://", "", ""}` if we think it's valid or maybe return an error in the apt table source code.
Though I would still test that case, together with another section in this test that tests failures for empty line, no comment, no protocol (possibly checking also that the error is what we expect).

Reply:
Sure thing, will improve the test coverage here.